forem/forem
# handle getElementById failures in articlePage

## What type of PR is this? (check all applicable)


## Description

Don't try to set the comment subscription content if there's no place to render it.

This is _known_ to occur for a small number of very old imported articles, but it's happening more frequently than that.

More importantly, don't try to write the innerHTML of a possibly null element in the error handling block, especially when the error might have been caused by a null response returned by getElementById (because the page changed while the code was running).

## Related Tickets & Documents

https://app.honeybadger.io/projects/67192/faults/79953220

## QA Instructions, Screenshots, Recordings

I had `console.log(e)` in the catch block, and focused the publishOrSavePost.spec.js:

```javascript
    it.only('Maintains draft status when publishing a draft fails', () => {
```

Run this a few times. If things went well, you'll see something like this (the asset hash 31f9 may differ)

```
TypeError: Cannot read properties of null (reading 'dataset')
    at eval (articlePage.jsx?31f9:103:27)
```

but not any test failure, which would show as this.

```
articlePage.jsx?31f9:131 Uncaught (in promise) TypeError: Cannot set properties of null (setting 'innerHTML')
    at eval (articlePage.jsx?31f9:131:5)
```

Run that e2e test until you confirm the console logged the handled error and the error handling did not fail/raise a new error.

### UI accessibility concerns?

None

## Added/updated tests?

- [x] No, and this is why: this is mostly covered by existing e2e tests (which were showing failures because of fast page changes).

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [x] This change does not need to be communicated, and this is why not: flaky spec fix
